microsoft-edge/devtools-guide-chromium/speed/get-started.md

```diff
@@ -22,440 +22,480 @@ ms.date: 05/04/2021
 
-It's time to see whether using production mode and reducing JavaScript activity may speed up the page load.  Start with production mode:
+    ![The Performance tool trace of the page load](./images/performance-trace.png)
 
-1. In the editor tab, open `webpack.config.js`.
+The trace shows activity chronologically, from left to right. The CPU and NET charts at the top give you an overview of CPU and network activity. The light brown area in the CPU chart corresponds to a period of time when the browser was busy with scripting activity. This area is a clue that you may be able to speed up page load by doing less JavaScript work.
 
-1. Change `"mode":"development"` to `"mode":"production"`.
+![A long scripting activity in the CPU chart](./images/perf-busy-script.png)
 
-1. Wait for the new build to deploy.
+Investigate the trace to find ways to do less JavaScript work.
 
-1. Audit the page again.
+The **Main** section shows a chronological log of main thread activity, from left to right.  The y-axis (top to bottom) shows why events occurred. For example, in the following figure, the browser spends most of the time in a **Parse HTML** event. As long as this event runs, the webpage isn't entirely displayed. This event caused a long **Evaluate Script** event, which corresponds to when the browser executes JavaScript code. The **Evaluate Script** event caused an `(anonymous)` function to run, which caused the `initRatings` function to run:
 
-   ![An Audits report after configuring webpack to use production mode](../media/speed-glitch-tony-remix-updated-3-audits-performance.msft.png)
+![Events in the Main section of the Performance tool](./images/perf-main-events.png)
 
-Reduce JavaScript activity by removing the request to `mineBitcoin`:
+Optimize the `initRatings` function to make the page load faster:
 
-1. In the editor tab, open `src/App.jsx`.
+1. In Visual Studio Code, open the `/travel-site/assets/rating.js` file and read through the code.
 
-1. Comment out the request to `this.mineBitcoin(1500)` in the `constructor`.
+    The code in this file is responsible for displaying five customer reviews on the webpage. Each review has a rating consisting of several stars, a quote, and an author. These reviews come from a server-side API used in the `getRatings` function, and the `initRatings` function creates DOM elements in the page for each review.
 
-1. Wait for the new build to deploy.
+1. In the `initRatings` function, find a way to make the JavaScript code faster.
 
-1. Audit the page again.
+    This webpage is just a demo and a `for` loop was introduced to slow down the code. In practice, there are many ways for a JavaScript file like this one to cause slow page loads, such as: accessing the DOM too frequently, running heavy computation tasks, using large JavaScript libraries or frameworks.
```

Comment:
```suggestion
    This webpage is just a demo, and a `for` loop was introduced to slow down the code. In practice, there are many ways for a JavaScript file like this one to cause slow page loads, such as: accessing the DOM too frequently, running heavy computation tasks, using large JavaScript libraries or frameworks.
```
opport. for comma, to deliver short separately delineated phrases b/c separate/distinct points.